--- a/common/bin-parser/parser/base/node.go
+++ b/common/bin-parser/parser/base/node.go
@@ -4,6 +4,7 @@
 	"errors"
 	"fmt"
 	"github.com/yaklang/yaklang/common/utils"
+	"github.com/yaklang/yaklang/common/utils/omap"
 	yaml "github.com/yaklang/yaklang/common/utils/orderedyaml"
 	"strconv"
 	"strings"
@@ -68,7 +69,7 @@
 }
 
 type BaseKV struct {
-	data *configStore
+	data *omap.OrderedMap[string, any]
 }
 
 func (c *BaseKV) DeleteItem(k string) {
@@ -139,7 +140,7 @@
 func NewEmptyConfig() *Config {
 	return &Config{
 		BaseKV: BaseKV{
-			&configStore{},
+			omap.NewEmptyOrderedMap[string, any](),
 		},
 	}
 }
@@ -176,7 +177,7 @@
 func NewConfig(config *Config) *Config {
 	res := &Config{
 		BaseKV: BaseKV{
-			&configStore{},
+			omap.NewEmptyOrderedMap[string, any](),
 		},
 	}
 	copeFields := []string{"endian", "parser", "unit"}
@@ -422,14 +423,14 @@
 func NewNodeTree(d yaml.MapSlice) (*Node, error) {
 	defaultConfig := &Config{
 		BaseKV: BaseKV{
-			&configStore{},
+			omap.NewEmptyOrderedMap[string, any](),
 		},
 	}
 	defaultConfig.SetItem("endian", "big")
 	defaultConfig.SetItem("parser", "default")
 	ctx := &NodeContext{
 		BaseKV: BaseKV{
-			&configStore{},
+			omap.NewEmptyOrderedMap[string, any](),
 		},
 	}
 	ctx.SetItem(ctxParserRuntimeMap, newParserRuntimeMap())
